kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ChecksumSimpleComparisonSuite.scala

```diff
@@ -0,0 +1,376 @@
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults
+
+import io.delta.kernel.Operation.CREATE_TABLE
+import io.delta.kernel.{Operation, Table, Transaction}
+import io.delta.kernel.data.{ColumnarBatch, Row}
+import io.delta.kernel.defaults.utils.TestUtils
+import io.delta.kernel.engine.Engine
+import io.delta.kernel.expressions.Literal
+import io.delta.kernel.internal.actions.{AddFile, SingleAction}
+import io.delta.kernel.internal.checksum.{CRCInfo, ChecksumReader}
+import io.delta.kernel.internal.fs.Path
+import io.delta.kernel.internal.util.{FileNames, VectorUtils}
+import io.delta.kernel.internal.util.Utils.{singletonCloseableIterator, toCloseableIterator}
+import io.delta.kernel.types.IntegerType.INTEGER
+import io.delta.kernel.types.StructType
+import io.delta.kernel.utils.CloseableIterable.{emptyIterable, inMemoryIterable}
+import io.delta.kernel.utils.{CloseableIterator, DataFileStatus, FileStatus}
+import org.scalatest.funsuite.AnyFunSuite
+
+import java.io.File
+import java.nio.file.Files
+import java.util
+import java.util.Collections.{emptyMap, singletonMap}
+import java.util.Optional
+import scala.collection.JavaConverters._
+
+/**
+ * Test suite to verify checksum file correctness by comparing
+ * Delta Spark and Delta Kernel generated checksum files.
+ */
+class ChecksumSimpleComparisonSuite extends AnyFunSuite with TestUtils {
+
+  private val PARTITION_COLUMN = "part"
+
+  test("create table, insert data and verify checksum") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val sparkTablePath = tablePath + "spark"
+      val kernelTablePath = tablePath + "kernel"
+      Table
+        .forPath(engine, kernelTablePath)
+        .createTransactionBuilder(engine, "test-engine", CREATE_TABLE)
+        .withSchema(engine, new StructType().add("id", INTEGER))
+        .build(engine)
+        .commit(engine, emptyIterable())
+        .getPostCommitHooks
+        .forEach(
+          hook => hook.threadSafeInvoke(engine)
+        )
+      spark.sql(s"CREATE OR REPLACE TABLE delta.`${sparkTablePath}` (id Integer) USING DELTA")
+      assertChecksumEquals(engine, sparkTablePath, kernelTablePath, 0)
+
+      (1 to 10).foreach(
+        version => insertIntoUnpartitionedTable(engine, sparkTablePath, kernelTablePath, version)
+      )
+
+    }
+  }
+
+  test("create table as select and verify checksum") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val sparkTablePath = tablePath + "spark"
+      val kernelTablePath = tablePath + "kernel"
+      spark.sql(
+        s"CREATE OR REPLACE TABLE delta.`${tablePath + "spark"}` USING DELTA AS SELECT 1 as id"
+      )
+      val txn = Table
+        .forPath(engine, kernelTablePath)
+        .createTransactionBuilder(engine, "test-engine", CREATE_TABLE)
+        .withSchema(engine, new StructType().add("id", INTEGER))
+        .build(engine)
+
+      copyAndCommitTxnForUnpartitionedTable(txn, engine, sparkTablePath, versionAtCommit = 0)
```

Comment:
I couldn't quite understand what this method is doing -- can you explain to me (on this PR, not in the code)?

Reply:
I was trying to make the test logic like

1.commit to spark table using ctas/dml, which produces a log with add file
2.read the log in step 1 and commit to the kernel table (make sore both table's log pointing to the same sets of parquet file to validate table size is correct in crc)
3.verify crc written stats are aligned

This method is mainly for step 2.

Does "copyDeltaLogFromSparkToKernel" a better name?

Reply:
Added some more documentation to the test class and method as follow up